File: src/main/java/com/illuzionzstudios/mist/conversation/SimplePrefix.java
package com.illuzionzstudios.mist.conversation;

import com.illuzionzstudios.mist.util.TextUtil;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.bukkit.conversations.ConversationContext;
import org.bukkit.conversations.ConversationPrefix;

/**
 * A simple conversation prefix with a static string
 */
@RequiredArgsConstructor
public final class SimplePrefix implements ConversationPrefix {

    /**
     * The conversation prefix
     */
    @Getter
    private final String prefix;

    @Override
    public String getPrefix(ConversationContext context) {
        return TextUtil.formatText(prefix);
    }
}

File: src/main/kotlin/com/illuzionzstudios/mist/conversation/SimplePrefix.kt
package com.illuzionzstudios.mist.conversation

import com.illuzionzstudios.mist.util.TextUtil
import com.illuzionzstudios.mist.util.TextUtil.formatText
import lombok.*
import org.bukkit.conversations.ConversationContext
import org.bukkit.conversations.ConversationPrefix

/**
 * A simple conversation prefix with a static string
 */
@RequiredArgsConstructor
class SimplePrefix : ConversationPrefix {
    /**
     * The conversation prefix
     */
    @Getter
    private val prefix: String? = null
    override fun getPrefix(context: ConversationContext): String {
        return TextUtil.formatText(prefix)
    }
}